lib/std/heap/memory_pool.zig

```diff
@@ -0,0 +1,183 @@
+const std = @import("../std.zig");
+
+const debug_mode = @import("builtin").mode == .Debug;
+
+pub const MemoryPoolError = error{OutOfMemory};
+
+/// A memory pool that can allocate objects of a single type very quickly.
+/// Use this instead of a general purpose allocator when you need to allocate a lot of objects
+/// of the same type, as a memory pool outperforms general purpose allocators.
+pub fn MemoryPool(comptime Item: type) type {
+    return MemoryPoolAligned(Item, @alignOf(Item));
+}
+
+/// A memory pool that can allocate objects of a single type very quickly.
+/// Use this instead of a general purpose allocator when you need to allocate a lot of objects
+/// of the same type, as a memory pool outperforms general purpose allocators.
+pub fn MemoryPoolAligned(comptime Item: type, comptime alignment: u29) type {
+    if (@alignOf(Item) == alignment) {
+        return MemoryPoolExtra(Item, .{});
+    } else {
+        return MemoryPoolExtra(Item, .{ .alignment = alignment });
+    }
+}
+
+pub const Options = struct {
+    /// The alignment of the memory pool items. Use `null` for natural alignment.
+    alignment: ?u29 = null,
+
+    /// If `true`, the memory pool can allocate additional items after a initial setup.
+    /// If `false`, the memory pool will not allocate further after a call to `initPreheated`.
+    growable: bool = true,
+};
+
+/// A memory pool that can allocate objects of a single type very quickly.
+/// Use this instead of a general purpose allocator when you need to allocate a lot of objects
+/// of the same type, as a memory pool outperforms general purpose allocators.
```

Comment:
Should the doc comments be somehow different between this and `MemoryPoolAligned` to indicate that they are different?

Reply:
Improved the doc comments for *Aligned and *Extra a bit